### PR TITLE
Apply platform universal offset in Player

### LINKED ISSUE #10: Slightly different pp results on Windows vs Linux/macOS
There seems to be some platform differences causing pp calculations to differ between Windows and Linux/macOS systems.

For example [Idke](https://osu.ppy.sh/u/4650315)'s score on https://osu.ppy.sh/b/1201636?m=0 results in ~720pp on Windows, and the expected ~718pp on Linux/macOS.

This is probably an issue with the .NET Core runtime itself and will require further investigation + possibly reporting to the dotnet team.

### LINKED ISSUE #2654: Platform-specific universal offsets must not be applied at decoding time
screen shot 2018-05-28 at 2 48 04 am

This needs to go. It causes differences in difficulty calculation.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/58899230-platform-specific-universal-offsets-must-not-be-applied-at-decoding-time?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by smoogipoo**
Resolves #2654. Fixes ppy/osu-tools#10.

Note: The offset direction has changed (-22 -> 22). This is intentional, because whereas previously we wanted hitobjects to be closer to 0, we now want the current time to be further from 0.

Also fixes the wrong clock being seeked when unpausing. Previously the non-offset clock would be seeked to the post-offset time, so the user would jump ahead/behind by their UO and the platform-UO introduced by this change.

Changed the ctor to use `DecoupleableInterpolatingFramedClock` for explanatory purposes.

### CODE REVIEW

**Comment by peppy** [comment]
Keep in mind this needs to be applied in the editor too going forward.

**Comment by smoogipoo** [comment]
At which point we'll make a base clock for `EditorClock` to derive and for `Player` to use.

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -143,8 +144,12 @@ private void load(AudioManager audio, APIAccess api, OsuConfigManager config)
...
 
+            // Lazer's audio timings in general doesn't match stable. This is the result of user testing, albeit limited.
+            // This only seems to be required on windows. We need to eventually figure out why, with a bit of luck.
+            var offsetClock = new FramedOffsetClock(adjustableClock) { Offset = RuntimeInfo.OS == RuntimeInfo.Platform.Windows ? 22 : 0 };
+
             // the final usable gameplay clock with user-set offsets applied.
-            var offsetClock = new FramedOffsetClock(adjustableClock);
+            offsetClock = new FramedOffsetClock(offsetClock);
 
             userAudioOffset.ValueChanged += v => offsetClock.Offset = v;
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: this is going to override the offset set above

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Play/Player.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game/Screens/Play/Player.cs]: There are two offset clocks.

Comment (peppy) [reply osu.Game/Screens/Play/Player.cs]: check kline above

**Review by peppy — APPROVED** [review]

**Pull request merged**